### PR TITLE
ci: prevent mutate method arguments by eslint

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Refactoring
-   [ ] Code style update
-   [x] Build or CI related changes
-   [ ] Documentation content changes

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/addon-commerce/components/input-card/input-card.component.ts [hunk 1]
```diff
@@ -38,9 +38,9 @@ const icons = {
 };
 
 export function cardTextfieldControllerFactory(
-    directive: TuiTextfieldExampleTextDirective | null,
+    value: TuiTextfieldExampleTextDirective | null,
 ): TuiTextfieldExampleTextDirective {
-    directive = directive || new TuiTextfieldExampleTextDirective();
+    let directive = value || new TuiTextfieldExampleTextDirective();
     directive.exampleText = '0000 0000 0000 0000';
```
Comment (waterplea) [inline projects/addon-commerce/components/input-card/input-card.component.ts:44]: Need an extra line around declaration. If there's no such rule in the new linter we need to add it.

Comment (splincode) [reply projects/addon-commerce/components/input-card/input-card.component.ts:44]: Not problem, in the next PR)

File: projects/addon-doc/src/components/navigation/navigation.component.ts [hunk 2]
```diff
@@ -117,11 +117,16 @@ export class TuiDocNavigationComponent {
...
         items: ReadonlyArray<ReadonlyArray<TuiDocPage>>,
-        search: string,
+        searchValue: string,
     ): ReadonlyArray<ReadonlyArray<TuiDocPage>> {
+        let search = searchValue;
+
         return items.map(section =>
             uniqBy(
-                section.filter(({title, keywords = ''}) => {
+                section.filter(({title: titleValue, keywords: keywordsValue}) => {
```
Comment (waterplea) [inline projects/addon-doc/src/components/navigation/navigation.component.ts:126]: I don't like complex destructs, let's rename variables below instead

File: projects/addon-editor/extensions/indent-outdent.ts [hunk 3]
```diff
@@ -140,7 +141,8 @@ export const Indent = Extension.create<IndentOptions>({
         return {
             indent:
                 () =>
-                ({tr, state, dispatch}) => {
+                ({tr: value, state, dispatch}) => {
```
Comment (waterplea) [inline projects/addon-editor/extensions/indent-outdent.ts:144]: Same here and below. Also unnecessary type.

**Pull request closed without merging**